dali/python/nvidia/dali/_utils/external_source_impl.py

```diff
@@ -120,14 +124,24 @@ def sample_to_numpy(sample, error_str="Unsupported callback return type. Got: `{
     raise TypeError(error_str.format(type(sample)))
 
 
-def batch_to_numpy(batch, error_str="Unsupported callback return type. Got: `{}`."):
+def batch_to_numpy(
+        batch,
+        error_str="Unsupported callback return type. Got: `{}`.",
+        check_uniform=True,
```

Comment:
Do we always need to check if the batch is uniform?
Does this applies only to TF here?

Reply:
I can revert the condition, it's only used in TF now. Do you want me to?

Reply:
I just wonder if by accident it won't be used by the regular ES where we allow this.

Reply:
I think there's some flaw in the design of this function to start with. If the input is a TensorListGPU, we try to stack the samples into one TensorListGPU - but if the input is a list, we return a list of numpy arrays. Is it really what we want?
Should we rather have an argument that controls whether we stack or not and then we'd just forbid stacking of non-uniform data no matter what?
Additionally, we could require the batch to be uniform even for non-stacked case - I don't know if that would be of value here.

Reply:
I made it stack unconditionally, with customizable error message.